src/rule.hpp

```diff
@@ -54,6 +54,13 @@ class condition
         const ddwaf::manifest &manifest, bool run_on_new,
         ddwaf::timer& deadline) const;
 
+    std::string_view processor_name() {
+        return processor_ ? processor_->name() : "";
```

Comment:
Processor may have changed between the check and the call to `name`. I realize this doesn't make a lot of difference because the type of the processor should be the same even after data updates (and if it did, you'd have bigger problems, because the type could have changed again afterwards). But still, by accessing the shared pointer in this fashion you have no guarantees on the visibility of the data behind the pointer. It should be something like:

```c++
return std::atomic_load_explicit(
        &processor_, std::memory_order_acquire)->name();
```

(assuming `std::memory_order_release` on the store side).